suite-native/device-manager/src/components/AddHiddenWalletButton.tsx

```diff
@@ -1,36 +1,23 @@
-import { useNavigation } from '@react-navigation/native';
+import { useDispatch, useSelector } from 'react-redux';
 
 import { Button } from '@suite-native/atoms';
 import { Translation } from '@suite-native/intl';
-import {
-    PassphraseStackParamList,
-    PassphraseStackRoutes,
-    RootStackRoutes,
-    StackToStackCompositeNavigationProps,
-    RootStackParamList,
-} from '@suite-native/navigation';
+import { createDeviceInstance, selectDevice } from '@suite-common/wallet-core';
 
 import { useDeviceManager } from '../hooks/useDeviceManager';
 
-type NavigationProp = StackToStackCompositeNavigationProps<
-    PassphraseStackParamList,
-    PassphraseStackRoutes.PassphraseForm,
-    RootStackParamList
->;
-
 export const AddHiddenWalletButton = () => {
-    const navigation = useNavigation<NavigationProp>();
+    const dispatch = useDispatch();
+
+    const device = useSelector(selectDevice);
 
     const { setIsDeviceManagerVisible } = useDeviceManager();
 
     const handleAddHiddenWallet = () => {
-        setIsDeviceManagerVisible(false);
+        if (!device) return;
```

Comment:
will it work with locked device?

Reply:
This will be reworked when I have design for wallet switcher. I don't think it will and it will also not work with discovery running. Disabled state will be designed by Sheny soon.